src/Microsoft.OData.Edm/Csdl/CsdlJsonReaderSettings.cs

```diff
@@ -0,0 +1,72 @@
+//---------------------------------------------------------------------
+// <copyright file="CsdlJsonReaderSettings.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System.Collections.Generic;
+
+namespace Microsoft.OData.Edm.Csdl
+{
+    /// <summary>
+    /// Settings used when parsing CSDL-JSON document.
+    /// </summary>
+    public class CsdlJsonReaderSettings
+    {
+        internal static CsdlJsonReaderSettings Default = new CsdlJsonReaderSettings();
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="CsdlJsonReaderSettings"/> class.
+        /// </summary>
+        public CsdlJsonReaderSettings()
+        {
+            IsBracketNotation = false;
+            IncludeDefaultVocabularies = true;
+            IgnoreUnexpectedJsonElements = true;
+        }
+
+        /// <summary>
+        /// Gets/sets the referened models.
+        /// </summary>
+        public IEnumerable<IEdmModel> ReferencedModels { get; set; }
+
+        /// <summary>
+        /// Gets/sets a call back to provide the JSON reader for referenced CSDL-JSON.
+        /// </summary>
+        public CsdlJsonReaderFactory JsonReaderFactory { get; set;}

```

Comment:
>JsonReaderFactory [](start = 37, length = 17)

JsonReaderFactory is a bit vague (doesn't convey that it is reading schemas).  It's really a CsdlReaderFactory, JsonCsdlReaderFactory, or (maybe best) JsonSchemaReaderFactory.

Reply:
Changed to: JsonSchemaReaderFactory

---
In reply to: [429381657](https://github.com/OData/odata.net/pull/1775#discussion_r429381657) [](ancestors = 429381657)